pkg/api/serialization_test.go

```diff
@@ -247,12 +251,15 @@ func TestUnversionedTypes(t *testing.T) {
 			continue
 		}
 
+		versionDecodedObject := pair.EmptyCopy
 		// Make sure the versioned codec under test can decode
-		versionDecodedObject, err := runtime.Decode(testapi.Default.Codec(), unversionedJSON)
+		err = runtime.DecodeInto(testapi.Default.Codec(), unversionedJSON, versionDecodedObject)
 		if err != nil {
 			t.Errorf("%v: unexpected error: %v", obj, err)
 			continue
 		}
+		// DecodeInto will set the APIVersion to the version the object are registered with, which we don't care.
+		versionDecodedObject.GetObjectKind().SetGroupVersionKind(&unversioned.GroupVersionKind{})
```

Comment:
@smarterclayton I observed calling `runtime.Decode()` and `runtime.DecodeInto()` on unversioned objects will return different results, `DecodeInto()` will set the APIVersion and Kind, while `Decode()` doesn't, is this expected?